packages/components/src/font-size-picker/index.tsx

```diff
@@ -201,141 +192,145 @@ const UnforwardedFontSizePicker = (
-											size={ size }
-										>
-											{ __( 'Reset' ) }
-										</ResetButton>
-									</FlexItem>
-								) }
-							</Flex>
-						) }
-				</Controls>
-				{ withSlider && (
-					<RangeControl
+				{ ! shouldUseSelectControl && ! showCustomValueControl && (
+					<ToggleGroupControl
 						__nextHasNoMarginBottom={ __nextHasNoMarginBottom }
-						className="components-font-size-picker__custom-input"
-						label={ __( 'Custom Size' ) }
-						value={
-							isPixelValue && noUnitsValue
-								? Number( noUnitsValue )
-								: undefined
-						}
-						initialPosition={ fallbackFontSize }
+						label={ __( 'Font size' ) }
+						hideLabelFromVision
+						value={ value }
 						onChange={ ( newValue ) => {
-							onChange?.( hasUnits ? newValue + 'px' : newValue );
+							if ( newValue === undefined ) {
+								onChange?.( undefined );
+							} else {
+								onChange?.(
+									hasUnits ? newValue : Number( newValue )
+								);
+							}
 						} }
-						min={ 12 }
-						max={ 100 }
-					/>
+						isBlock
+						size={ size }
+					>
+						{ ( options as FontSizeToggleGroupOption[] ).map(
+							( option ) => (
+								<ToggleGroupControlOption
+									key={ option.key }
+									value={ option.value }
+									label={ option.label }
+									aria-label={ option.name }
+									showTooltip={ true }
+								/>
+							)
+						) }
+					</ToggleGroupControl>
+				) }
+				{ ! disableCustomFontSizes && showCustomValueControl && (
+					<Flex className="components-font-size-picker__custom-size-control">
+						<FlexItem isBlock>
+							<UnitControl
+								label={ __( 'Custom' ) }
+								labelPosition="top"
+								hideLabelFromVision
+								value={ value }
+								onChange={ ( newValue ) => {
+									if ( newValue === undefined ) {
+										onChange?.( undefined );
+									} else {
+										onChange?.(
+											hasUnits
+												? newValue
+												: parseFloat( newValue )
+										);
+									}
+								} }
+								size={ size }
+								units={ hasUnits ? units : [] }
+							/>
+						</FlexItem>
+						{ withSlider && (
+							<FlexItem isBlock>
+								<RangeControl
+									__nextHasNoMarginBottom={
+										__nextHasNoMarginBottom
+									}
+									className="components-font-size-picker__custom-input"
+									label={ __( 'Custom Size' ) }
+									hideLabelFromVision
+									value={ valueNumber }
+									initialPosition={ fallbackFontSize }
+									withInputField={ false }
+									onChange={ ( newValue ) => {
+										if ( newValue === undefined ) {
+											onChange?.( undefined );
+										} else if ( hasUnits ) {
+											onChange?.(
+												newValue + ( valueUnit ?? 'px' )
+											);
+										} else {
+											onChange?.( newValue );
+										}
+									} }
+									min={ isValueUnitRelative ? 0.75 : 12 }
+									max={ isValueUnitRelative ? 6.25 : 100 }
+									step={ isValueUnitRelative ? 0.05 : 1 }
```

Comment:
I chose these new `em` values to be roughly equivalent to the previous pixel values assuming a 16px base font size.

Reply:
These values feel nice to play with in the UI! 👍

Reply:
@glendaviesnz also made similar changes in https://github.com/WordPress/gutenberg/pull/44959 for spacing. Font size is a bit different to spacing, so not sure if it makes sense to align these values.